pkgs/extension_discovery/lib/extension_discovery.dart

```diff
@@ -128,8 +131,18 @@ Future<List<Extension>> findExtensions(
   String targetPackage, {
   bool useCache = true,
   Uri? packageConfig,
+  Uri? packageRootDir,
```

Comment:
I don't mind this, but I think it might be just as fine to create an auxiliary function:
```dart
/// Find [Uri] for `.dart_tool/package_config.json` searching up from [directory].
Uri? findPackageConfig(Uri directory) {
  ...
}
```

Then users can combine `findExtensions` and `findPackageConfig` if they want to search from a arbitrary location.

We don't even care if it's the package root or not.

Reply:
Done